subprojects/build-events/src/main/java/org/gradle/internal/build/event/types/DefaultFailure.java

```diff
@@ -72,38 +70,38 @@ public List<InternalBasicProblemDetailsVersion3> getProblems() {
     }
 
     public static InternalFailure fromThrowable(Throwable throwable) {
-        return fromThrowable(throwable, t -> ImmutableList.of(), p -> null);
+        return fromThrowable(throwable, p -> null);
     }
 
-    public static InternalFailure fromThrowable(Throwable t, ProblemLocator problemLocator, Function<InternalProblem, InternalBasicProblemDetailsVersion3> mapper) {
-
-        // Iterate through the cause hierarchy, with including multi-cause exceptions and convert them to a corresponding Failure with the same cause structure. If the current exception has a
-        // corresponding problem in `problemsMapping` (ie the exception was thrown via ProblemReporter.throwing()), then the problem will be also available in the new failure object.
-        StringWriter out = new StringWriter();
-        PrintWriter wrt = new PrintWriter(out);
-        t.printStackTrace(wrt);
-        Throwable cause = t.getCause();
-        List<InternalFailure> causeFailures = getCauseFailures(problemLocator, mapper, cause);
-        List<InternalBasicProblemDetailsVersion3> problemMapping =
-            problemLocator.findAll(t)
-                .stream()
-                .filter(Objects::nonNull)
-                .map(mapper)
-                .filter(Objects::nonNull)
-                .collect(toList());
-
-        return new DefaultFailure(t.getMessage(), out.toString(), causeFailures, problemMapping);
+    public static InternalFailure fromThrowable(Throwable t, Function<InternalProblem, InternalBasicProblemDetailsVersion3> mapper) {
+        Failure failure = DefaultFailureFactory.withDefaultClassifier().create(t);
```

Comment:
@gradle/bt-ide-experience How much sense would it make to pass the FailureFactory into this method instead of creating a new failure factory here? That would probably mean that well need to inject the service into the caller.

Reply:
Instead of passing the factory, we should consider getting rid of overloads that take a Throwable and accept a Failure instead. The usage of the factory to produce one from the other will be a call-site concern.